proposals/lip-0044.md

```diff
@@ -594,32 +595,45 @@ def getGeneratorsBetweenTimestamps(startTimestamp: uint32, endTimestamp: uint32)
     return result
 ```
 
-#### setGeneratorList
+#### setValidatorParams
 
-This function updates the value of the generator list to the input `generatorAddresses`. It checks if all elements of `generatorAddresses` have a registered validator account in the validators data substore. If this is not the case, it throws an error. Otherwise, it updates the value of `generatorList`.
+This function allows to set the precommit threshold, certificate threshold, validators and associated BFT weights to be used from the next height onward. The information is not validated and simply stored in the Validators module. The information is further forwarded to the engine where all validations are performed.
```

Comment:
I added the following line to the function:
```
forward params to consensus domain
```

What happens in the engine if the validation fails, is now defined in the section "After Application Processing" in LIP 58.

Reply:
ok. It would be good to reference the sections from LIP 58 as it's very hard for a reader of this section to see where in LIP 58 this data is processed /validated.
Maybe it's also not very obvious that the consensus domain has to keep the forwarded data in memory until it a certain stage (after application processing) is reached and will not process/validate it directly.

One idea: LIP 0058 introduces an auxiliary data structure `aux_BFT_params` that is kept in memory and that is filled by a function `set_aux_BFT_params` which is called by `setValidatorParams`. Then in  After Application Processing `setBFTParameters`is called with `aux_BFT_params` if `aux_BFT_params` not empty. At the end of After Application Processing is cleared such that it is empty once the next block is processed.

Reply:
I added some links to the function description to make it more clear what happens.

I would not add any auxiliary data structure for temporary storage or get very detailed regarding the implementation. @shuse2  was explaining that the transfer of information basically works as follows:
```
const result = abi.afterTransactionsExecute();
```
If I understand correctly here the application is called from the consensus domain via the application-blockchain-interface. What "forward params to consensus domain" is to add the BFT parameters to the return value of the function `afterTransactionsExecute`. The `result` value is then used in the after application processing stage within the consensus domain.